flake.nix

```diff
@@ -3,32 +3,42 @@
   
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+    circom-compat.url = "github:codex-storage/circom-compat-ffi";
   };
 
-  outputs = { self, nixpkgs }:
+  outputs = { self, nixpkgs, circom-compat}:
     let
-      supportedSystems = [
+      stableSystems = [
         "x86_64-linux" "aarch64-linux"
         "x86_64-darwin" "aarch64-darwin"
       ];
-      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
-      pkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+      forEach = nixpkgs.lib.genAttrs;
+      forAllSystems = forEach stableSystems;
+      pkgsFor = forEach stableSystems (
+        system: import nixpkgs { inherit system; }
+      );
     in rec {
+      packages = forAllSystems (system: let
+        circomCompatPkg = circom-compat.packages.${system}.default;
+        buildTarget = pkgsFor.${system}.callPackage ./nix/default.nix {
+          inherit stableSystems circomCompatPkg;
+          src = self;
+        };
+        build = targets: buildTarget.override { inherit targets; };
+      in rec {
+        codex   = build ["all"];
+        default = codex;
```

Comment:
One weird thing I found when testing was that if I specify target with `#codex` or `#default` it builds, but if I leave that out then it fails with:
```
 > nix build --print-build-logs --print-out-paths '.?submodules=1'
warning: Git tree '/home/jakubgs/work/nim-codex' is dirty
codex> Running phase: unpackPhase
codex> unpacking source archive /nix/store/k9ri8k88313z6ggkcm4k95aaapqw351i-source
codex> source root is source
codex> Running phase: patchPhase
codex> Running phase: updateAutotoolsGnuConfigScriptsPhase
codex> Running phase: configurePhase
codex> Running phase: buildPhase
codex> build flags: SHELL=/nix/store/516kai7nl5dxr792c0nzq0jp8m4zvxpi-bash-5.2p32/bin/bash all V=2 USE_SYSTEM_NIM=1
codex> Git submodules not found. Running 'git submodule update --init --recursive'.
codex> 0.1.0-dirty
codex> make: Nothing to be done for 'all'.
codex> Running phase: installPhase
codex> cp: cannot stat 'build/codex': No such file or directory
```
Surprisingly `.?submodules=1#` works fine, so that smells like a nix interpreter bug that ignores the submodules arg.